core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -170,29 +181,22 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
 
   private def recoverIndexInternal(checkpointMillis: Option[Long]): Future[Long] = {
     // By passing -1 for partId, numeric partId will not be persisted in the index
+    // Since we are recovering, always start from last synced time and update till current timestamp.
+    val endHour = hour()
+
     (checkpointMillis match {
       case Some(tsMillis)   => // We know we have to refresh only since this hour, it is possible we do not
                                 // find any data for this refresh as the index is already updated
-                                val (startHour, endHour) = (hour(tsMillis), hour())
+                                val startHour = hour(tsMillis) - 1
```

Comment:
Is the -1 here is added to avoid rounding errors and to make sure we are always looking back ?

Reply:
Yes, its to account for cases where the last refresh happened exactly at the same time we snapped the time last time.